packages/react-dev-utils/warnBrowsers.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+'use strict';
+
+const chalk = require('chalk');
+const findUp = require('find-up');
+const fs = require('fs');
+const path = require('path');
+
+function findConfig(dir) {
+  if (process.env.BROWSERSLIST) {
+    return '$BROWSERSLIST';
+  }
+
+  let file;
+
+  do {
+    const _file = findUp.sync(
+      ['browserslist', '.browserslistrc', 'package.json'],
+      {
+        cwd: dir,
+      }
+    );
+
+    if (_file == null) {
+      break;
+    }
+    dir = path.resolve(path.join(path.dirname(_file), '..'));
+
+    if (path.basename(_file) === 'package.json') {
+      try {
+        if (JSON.parse(fs.readFileSync(_file)).browserslist != null) {
+          file = _file;
+        }
+      } catch (_) {
+        // ignored
+      }
+    } else {
+      file = _file;
+    }
+  } while (file == null);
+
+  return file;
+}
+
+function getBrowsersWarning(dir) {
+  const found = findConfig(dir);
+
+  const instruction = `Please add a ${chalk.underline(
+    'browserslist'
+  )} key to your ${chalk.bold('package.json')}.`;
+
+  if (found == null) {
+    return {
+      error: true,
+      message: chalk.red('You must specify targeted browsers. ' + instruction),
+    };
+  }
+
+  if (found === '$BROWSERSLIST') {
+    return {
+      error: false,
+      message: chalk.yellow(
+        '$BROWSERSLIST was specified in your environment. ' +
```

Comment:
I'm not feeling great about being patronizing here.

If we decided to use `browserlists` then IMO we should embrace however they're doing it. I think our tone should be more like

"Built the bundle with support for <list of browsers> because <where we read the list>"

Reply:
Do we care about where the configuration comes from, or just to display what configuration was used?

Reply:
It would be nice to display both. But if that requires reimplementing half of browserlist maybe we can just show the config used, and file an issue with them asking to expose the source info.